src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

```diff
@@ -0,0 +1,802 @@
+        public bool CanPauseAndContinue
+        {
+            get
+            {
+                GenerateStatus();
+                return (_commandsAccepted & Interop.ACCEPT_PAUSE_CONTINUE) != 0;
+            }
+        }
+
+
+        /// Tells if the service is notified when system shutdown occurs.
+        public bool CanShutdown
+        {
+            get
+            {
+                GenerateStatus();
+                return (_commandsAccepted & Interop.ACCEPT_SHUTDOWN) != 0;
+            }
+        }
+
+        /// Tells if the service referenced by this object can be stopped.
+        public bool CanStop
+        {
+            get
+            {
+                GenerateStatus();
+                return (_commandsAccepted & Interop.ACCEPT_STOP) != 0;
+            }
+        }
+
+        /// The descriptive name shown for this service in the Service applet.
+        public string DisplayName
+        {
+            get
+            {
+                if (_displayName == null)
+                    GenerateNames();
+                return _displayName;
+            }
+        }
+
+        /// The set of services that depend on this service. These are the services that will be stopped if
+        /// this service is stopped.
+        public ServiceController[] DependentServices
+        {
+            get
+            {
+                if (_dependentServices == null)
+                {
+                    IntPtr serviceHandle = GetServiceHandle(Interop.SERVICE_ENUMERATE_DEPENDENTS);
+                    try
+                    {
+                        // figure out how big a buffer we need to get the info
+                        int bytesNeeded = 0;
+                        int numEnumerated = 0;
+                        bool result = Interop.mincore.EnumDependentServices(serviceHandle, Interop.SERVICE_STATE_ALL, IntPtr.Zero, 0,
+                            ref bytesNeeded, ref numEnumerated);
+                        if (result)
+                        {
+                            _dependentServices = new ServiceController[0];
+                            return _dependentServices;
+                        }
+
+                        int lastError = Marshal.GetLastWin32Error();
+                        if (lastError != Interop.ERROR_MORE_DATA)
+                            throw new Win32Exception(lastError);
+
+                        // allocate the buffer
+                        IntPtr enumBuffer = Marshal.AllocHGlobal((IntPtr)bytesNeeded);
+
+                        try
+                        {
+                            // get all the info
+                            result = Interop.mincore.EnumDependentServices(serviceHandle, Interop.SERVICE_STATE_ALL, enumBuffer, bytesNeeded,
+                                ref bytesNeeded, ref numEnumerated);
+                            if (!result)
+                                throw new Win32Exception(Marshal.GetLastWin32Error());
```

Comment:
This is equivalent to just `throw new Win32Exception();' since win32Exception's parameterless ctor calls Marshal.GetLastWin32Error().